### PR TITLE
Also add coverage job/tab if no periodic job defined

### LINKED ISSUE #615: make_config.go misses coverage job/tab if no periodic job defined
In https://github.com/knative/test-infra/pull/614 the knative/client repo has presubmit tests, but no periodic jobs defined yet (e.g., CI flow). The generator misses adding the coverage tab and the daily coverage job, since the presubmit section is empty.

/assign @Fredy-Z

### PR DESCRIPTION
**Pull request by Fredy-Z**
Fixes #615

### CODE REVIEW

**Comment by srinivashegde86** [comment]
did you run make config and there are no changes to the config?

**Comment by Fredy-Z** [comment]
> did you run make config and there are no changes to the config?

Yes, there are no changes to the config after I run `make config`.

File: ci/prow/make_config.go [hunk 1]
```diff
@@ -1046,6 +1045,17 @@ func parseSection(config yaml.MapSlice, title string, generate sectionGenerator,
 	}
 }
 
+// generateOtherJobConfigs generates job config with the generator if new job is required for it.
+func generateOtherJobConfigs(title string, newJobNeeded newJobNeeded, generate sectionGenerator) {
+	for i := range repositories { // Keep order for predictable output.
+		if !newJobNeeded(repositories[i]) {
+			continue
+		}
+
```
Comment (adrcunha) [inline ci/prow/make_config.go]:
```suggestion
```

File: ci/prow/make_config.go [hunk 2]
```diff
@@ -1311,6 +1329,7 @@ func addTestCoverageJobIfNeeded(jobDetailMap *map[string][]string, repoName stri
 	if goCoverageMap[repoName] {
 		newJobTypes := append((*jobDetailMap)[repoName], "test-coverage")
 		(*jobDetailMap)[repoName] = newJobTypes
+		delete(goCoverageMap, repoName)
```
Comment (adrcunha) [inline ci/prow/make_config.go]: Please comment explaining why you need to remove the repo from the map.

File: ci/prow/make_config.go [hunk 3]
```diff
@@ -1553,10 +1572,15 @@ func main() {
 		}
 		parseSection(config, "presubmits", generatePresubmit, nil)
 		parseSection(config, "periodics", generatePeriodic, generateGoCoveragePeriodic)
+		generateOtherJobConfigs("periodics", func(repo repositoryData) bool {
+			return !repo.Processed && repo.EnableGoCoverage
```
Comment (adrcunha) [inline ci/prow/make_config.go]: Do you really need .Processed? It seems to me that you're iterating only once over the repos...

Comment (Fredy-Z) [reply ci/prow/make_config.go]: Yes we need.
In the loop of `parseSection`, we call `generateGoCoveragePeriodic` to process all repos that have both `go-coverage` config in `presubmits` section and config in `periodics` section.
In the loop of `generateOtherJobConfigs`, we call `generateGoCoveragePeriodic` to process all others, that is - all repos that have `go-coverage` config in `presubmits` section but no config in `periodics`.

File: ci/prow/make_config.go (hunk 2 above)
Comment (Fredy-Z) [reply ci/prow/make_config.go]: Done.

Comment (Fredy-Z) [reply ci/prow/make_config.go]: And moved the logic to process other repos into a new function.

**Comment by adrcunha** [comment]
/approve

Please fix the merge conflict.

**Pull request merged**